Dockerfile for web

Romeo509 opened the pull request.

Romeo509 committed:
Dockerfile for web

ocansey11 committed:
Review: Removed .env, .env.example, docker-README.md, docker-compose.yml, nginx.conf per PR #3 review

ocansey11 committed:
Review: Changed the Dockerfile to be much simpler for development.After a little research i realized nginx was overkill. Since we are currently in development we will stick to lightweight settings

ocansey11 committed:
Review: One docker file is enough. I removed Docekr.web

ocansey11 committed:
I moved the Dockerfile inside the web folder

ocansey11 committed:
feat: addeded docker-compose file

ocansey11 committed:
feat: addeded docker file for server

ocansey11 reviewed (approved):
Made changes. Added Docker for Server and Web. As well as the docker-compose. Ill try it out on my local machine and see if it will build and automatically setup everything.

ocansey11 committed:
Updated docker-compose and added setup.sh for automation

ocansey11 committed:
feat: Add Docker setup with automated Better Auth secret generation

- Add Dockerfiles for web and server applications
- Add docker-compose.yml with environment variable configuration
- Add setup scripts (setup.sh and setup.bat) for cross-platform support
- Automated Better Auth secret generation and .env file creation
- Database migrations handled automatically in server container
- Remove unused root Dockerfile

Setup now requires only: ./setup.sh or setup.bat

ocansey11 committed:
added pirsma studio and tunneled for port 5555

tedashikode reviewed (approved):
The Docker setup doesn't work for me entirely, I'm running the Ubuntu OS, version 24.04.

 I changed the docker-compose command in the setup.sh bash script to docker compose but the secret key failed to get generated. I generated the secret key separately as @ocansey11 suggested. 
 
 However, I get the error in the screenshot when I try to run the Docker images(please correct me if I'm using the wrong terms) separately for the web and server directories.
 
 Kevin and I tried our best to fix the error but we couldn't figure it out. It seems to only occur on my machine so I'll approve the PR as everything else works fine - starting the dev servers for the frontend and backend works. 

<img width="1920" height="1080" alt="Screenshot from 2025-08-08 11-52-28" src="https://github.com/user-attachments/assets/b63896a7-32fe-49e8-a06c-a1d85fa2d021" />
